homeassistant/components/spc/__init__.py

```diff
@@ -1,81 +1,115 @@
 """Support for Vanderbilt (formerly Siemens) SPC alarm systems."""
 
+from __future__ import annotations
+
 import logging
+from typing import cast
 
+from aiohttp import ClientError
 from pyspcwebgw import SpcWebGateway
 from pyspcwebgw.area import Area
 from pyspcwebgw.zone import Zone
 import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers import aiohttp_client, discovery
-import homeassistant.helpers.config_validation as cv
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, device_registry as dr
 from homeassistant.helpers.dispatcher import async_dispatcher_send
 from homeassistant.helpers.typing import ConfigType
 
-_LOGGER = logging.getLogger(__name__)
-
-CONF_WS_URL = "ws_url"
-CONF_API_URL = "api_url"
-
-DOMAIN = "spc"
-DATA_API = "spc_api"
+from .const import (
+    CONF_API_URL,
+    CONF_WS_URL,
+    DATA_SCHEMA,
+    DOMAIN,
+    SIGNAL_UPDATE_ALARM,
+    SIGNAL_UPDATE_SENSOR,
+)
 
-SIGNAL_UPDATE_ALARM = "spc_update_alarm_{}"
-SIGNAL_UPDATE_SENSOR = "spc_update_sensor_{}"
+_LOGGER = logging.getLogger(__name__)
 
 CONFIG_SCHEMA = vol.Schema(
     {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_WS_URL): cv.string,
-                vol.Required(CONF_API_URL): cv.string,
-            }
-        )
+        DOMAIN: DATA_SCHEMA,
```

Comment:
I strongly recommend to keep the data schema here as that way we don't touch it and it will be removed in a while anyway